File: lab_solutions/lab2task2/scripts/subscriber_node.py
#!/usr/bin/env python
import rospy
import numpy as np
##TODO: include the library for a msg for an array integer
from std_msgs.msg import Int64MultiArray

##TODO: Complete the callback function. Your callback function should compute the sum (1 + 1/2 + 1/3 +... + 1/n) and print the output on the screen

def callback(msg):
	 
	my_array=msg.data
	
	sum=0
	for i in range(len(my_array)):
		if (my_array[i] != 0):
			sum+=(1/float(my_array[i]))
	print str(sum)

def listener():

    rospy.init_node('listener', anonymous=True)

    ##TODO: Define your subscriber and link this with the callback function.
    rospy.Subscriber("chatter", Int64MultiArray, callback)
    rospy.spin()


if __name__ == '__main__':
    listener()

File: lab_solutions/lab2task2/src/subscriber_node.cpp
#include "ros/ros.h"
//TODO: include the library for a msg for an array integer
#include "std_msgs/Float64MultiArray.h"

//TODO: Complete the callback function. Your callback function should compute the sum (1 + 1/2 + 1/3 +... + 1/n) and print the output on the screen
void subscribeCallback(const std_msgs::Float64MultiArray::ConstPtr& msg){
    double sum = 0;

    for (std::vector<double>::const_iterator it = msg->data.begin(); it != msg->data.end(); ++it)
	sum = sum + 1.0 / *it;

    printf("Current sum is: %.4f\n", sum);
};

int main(int argc, char **argv)
{
    ros::init(argc, argv, "listener");

    //TODO: Define your subscriber and link this with the callback function.
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("/chatter", 1000, subscribeCallback);

    ros::spin();

    return 0;
}
